gb-cpu/src/cpu.rs

```diff
@@ -1,5 +1,62 @@
 use crate::registers::Registers;
+use gb_bus::{AddressBus, Bus};
+use gb_clock::Ticker;
+use gb_roms::{Opcode, OpcodeGenerator};
 
+#[derive(Default)]
 pub struct Cpu {
-    _registers: Registers,
+    registers: Registers,
+}
+
+impl Cpu {
+    fn execute_opcode(&mut self, opc: Opcode, _addr_bus: &mut AddressBus) {
+        log::error!("Unhandled opcode: {:?}", opc);
+    }
+}
+
+impl Ticker<AddressBus> for Cpu {
+    fn cycle_count(&self) -> gb_clock::Tick {
+        gb_clock::Tick::MCycle
+    }
+
+    fn tick(&mut self, addr_bus: &mut AddressBus) {
+        let it = CpuIterator::new(self, addr_bus);
+        let mut gen = OpcodeGenerator::new(it);
+        if let Some(next) = gen.next() {
+            next.map_or_else(
+                |e| log::warn!("OpcodeGenerator returned an error: {:?}", e),
```

Comment:
why not logging an error here instead of a warning ?

Reply:
For me an error is something that should stop the current task.
In the cpu context, all error are ignored that's why I use a `warning` level instead of a `error`